### ISSUE TITLE
[Proposal] Add `params` argument to `FuncEnv`

### ISSUE THREAD

**Issue by Kallinteris-Andreas** [Farama-Foundation/Gymnasium#905]
### Proposal

Currently, the `__init__` function does not take the `params` argument, instead it takes the `options` argument

https://github.com/Farama-Foundation/Gymnasium/blob/f88a61852b50fed3c907e2104019f453b9044c3a/gymnasium/functional.py#L49-L52

But the `params` of the environment are needed to initialize the object (such as to set `observation_space` member variable)

additional changes requirements:
`get_default_params` should be changed to not require the instantiation of the environment to be used as it will be required in `__init__`
https://github.com/Farama-Foundation/Gymnasium/blob/de909da30a1fe40288275da331fa7ad8e219a2bc/gymnasium/functional.py#L123-L125
to
```py
 def get_default_params(**kwargs) -> Params | None: 
     """Get the default params.""" 
     return None
```
and update `FunctionalJaxEnv(gym.Env)` with that change

@RedTachyon 

### Motivation

I have been developing the [MJX based environments](https://github.com/Farama-Foundation/Gymnasium/issues/833) and this is a requirement

### Additional context

Here are 2 examples of `params` I need for MJX
1) `xml_file` indicates the file of the robot model to be used and sets up the internal MuJoCo model and everything else that depends on it like action and observation spaces

2) `exclude_current_positions_from_observation` changes, how much of the state space is observable, it is need in `__init__()` to `observation_space` and later in `observation()`

### Checklist

- [X] I have checked that there is no similar [issue](https://github.com/Farama-Foundation/Gymnasium/issues) in the repo

**Comment by pulak-gautam** [comment]
Since 1.0 has been released, will this be worked on? I am in the same boat (writing MJX envs). Is there any workaround for now?

**Comment by pulak-gautam** [comment]
Yes, more than happy to! I'll start working on it and submit the PR shortly
